src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c

```diff
@@ -9,13 +9,43 @@
 #define FLAGS 0x0A000011
 
 #define THIS ((ObjTimeblock*)thisx)
+#define BOOL(x) (x ? 1 : 0)
```

Comment:
yea agreed that you shouldnt need this

Reply:
It unfortunately affects matching. Let me give you one example. If I apply this:

```patch
--- a/src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c
+++ b/src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c
@@ -149,7 +149,7 @@ void ObjTimeblock_Init(Actor* thisx, GlobalContext* globalCtx) {
 
     Actor_SetHeight(&this->dyna.actor, sSizeOptions[(this->dyna.actor.params >> 8) & 1].height);
 
-    this->unk_174 = BOOL(Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F));
+    this->unk_174 = !!(Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F));
     this->unk_175 = BOOL((this->dyna.actor.params >> 15) & 1);
     this->isVisible = ObjTimeblock_CalculateIsVisible(this);
```

I get this:

<pre>274:    lh      a1,0x1c(s0)                         274:    lh      a1,0x1c(s0)
278:    lw      a0,<font color="#FCE94F">0x3c(sp)</font>                       <font color="#C4A000">s </font>278:    lw      a0,<font color="#FCE94F">0x44(sp)</font>
27c:    jal     Flags_GetSwitch                     27c:    jal     Flags_GetSwitch
280:    andi    a1,a1,0x3f                          280:    andi    a1,a1,0x3f
<font color="#CC0000">284:    beqz    v0,294</font> <font color="#CC0000">~&gt;</font>                         <font color="#CC0000">&lt; </font>
<font color="#729FCF">288:    li      t1,1</font>                              <font color="#729FCF">| 284:    lh      t2,0x1c(s0)</font>
<font color="#729FCF">28c:    b       298</font> <font color="#FCE94F">~&gt;</font>                            <font color="#729FCF">| 288:    sltu    t1,zero,v0</font>
290:    sb      t1,0x174(s0)                        28c:    sb      t1,0x174(s0)
<font color="#CC0000">294: ~&gt; sb      zero,0x174(s0)</font>                    <font color="#CC0000">&lt; </font>
<font color="#CC0000">298: </font><font color="#FCE94F">~&gt;</font><font color="#CC0000"> lh      t2,0x1c(s0)</font>                       <font color="#CC0000">&lt; </font>
<font color="#CC0000">29c:    li      t5,1</font>                              <font color="#CC0000">&lt; </font>
2a0:    sra     t3,t2,0xf                           290:    sra     t3,t2,0xf
2a4:    andi    t4,t3,0x1                           294:    andi    t4,t3,0x1</pre>

I also tried removing all `BOOL` and replacing them with `!!` or not having them altogether, to no avail. This module is riddled with these `beqz` + `b` patterns, and the only way I found to get them to match is by either wrapping them in a ternary, or by writing if-else statements.

Since it's so prevalent and makes the file more difficult to read, I wrote a macro. I can rename the macro to something more specific if that would help? Or maybe there's another pattern I am not aware of?

Reply:
I wouldn't oppose this macro if it was named like `TIMEBLOCK_BOOL` but then it isn't more readable than `(...) ? 1 : 0`
This file is very weird with the xors and that ternary

I think it's fine to use the ternary everywhere. (and forget about using true/false like ` ? true : false` to prevent clutter)